plugin/restore/mysql/mysql.go

```diff
@@ -693,6 +705,53 @@ func (r *Restore) CheckEngineInnoDB(ctx context.Context, database string) error
 	return nil
 }
 
+func (r *Restore) getServerVariable(ctx context.Context, varName string) (string, error) {
+	db, err := r.driver.GetDbConnection(ctx, "")
+	if err != nil {
+		return "", err
+	}
+
+	rows, err := db.QueryContext(ctx, fmt.Sprintf("SHOW VARIABLES LIKE '%s';", varName))
+	if err != nil {
+		return "", err
+	}
+	if ok := rows.Next(); !ok {
+		return "", fmt.Errorf("SHOW VARIABLES LIKE '%s' returns empty set", varName)
+	}
+	var varNameFound, value string
+	if err := rows.Scan(&varNameFound, &value); err != nil {
+		return "", err
+	}
+	if varName != varNameFound {
+		return "", fmt.Errorf("expecting variable %s, but got %s", varName, varNameFound)
+	}
+	return strings.ToUpper(value), nil
```

Comment:
This seems a bit magic. getServerVariable() should just return the value exactly returned by the server because some callers may want the original values. It's better to let the caller handle it.